core/tx_pool.go

```diff
@@ -1171,20 +1197,35 @@ func (pool *TxPool) demoteUnexecutables() {
 		for _, tx := range drops {
 			hash := tx.Hash()
 			logger.Warn().Str("hash", hash.Hex()).Msg("Removed unpayable pending transaction")
+			erroredTxns = append(erroredTxns, types.RPCTransactionError{
+				TxHashID:             hash.Hex(),
+				TimestampOfRejection: time.Now().Unix(),
+				ErrMessage:           "unpayable pending transaction",
+			})
```

Comment:
Same here, add the balances or gas cost. Basically we want the user to get least confused.

Reply:
Added informative msgs in latest